Tests/Integration/Network/UrlLoadBalancerIntTests/UrlLoadBalancerIntTests.swift

```diff
@@ -97,20 +198,45 @@ class UrlLoadBalancerIntTests: XCTestCase {
             }
             expectFailure.fulfill()
         }
-        waitForExpectations(timeout: expectationTimeout)
+        waitForExpectations(timeout: timeToWait)
+
+        // verify that rotate has been called by checking curIdx of the loadBalancer
+        XCTAssertEqual( 1, consensusUrlLoadBalancerMock.curIdx, "Consensus URL load balancer should have rotated from index 0 to 1")
+    }
+
+    func testConsensusDoesNotRotateAwayFromGoodUrl(using transportProtocol: TransportProtocol, andWaitFor timeToWait: Double) throws {
+        print("\n-----\nTesting Consensus url rotation does not occur with good URL for protocol: \(transportProtocol.description)")
+
+        let fogUrlLoadBalancer = try createValidFogUrlLoadBalancer()
```

Comment:
I think all of these test-configuration lines should be moved into a "fixtures" file. There a bunch of examples if you search the repo, but the idea is to make an internal, named struct for each config permutation thats needed in testing, then you instantiate that struct and use the preconfigured bits in the test. It simplifies the test functions and makes things more readable.